### DISCUSSION TITLE
Versioning and breaking api changes

### DISCUSSION THREAD

**Discussion by AudioDan** [futzu/threefive#24] [General]
First of all great product!  Having a lot of fun playing with it.

I would like to suggest that a little more care may be needed surrounding versioning in the setup files.  For example, we were testing with 2.2.45.  In the setup.py file, that dictates bitn>=0.0.33.  However in 0.0.41 of bitn, there is an api change from asdecodedhex to asascii.  This effectively breaks 2.2.45 on a fresh install when parsing some messages (including the splice message in the xaa demo video file).

So either threefive setup should pin to a specific bitn version rather than >= or bitn needs a major semver bump for breaking api changes.

Thanks - and again great product.

**Comment by futzu** [comment]
AudioDan, 
I'm glad you like it and I apologize for the version/ bitn api issue. 
I just came up with the even number version idea a few days ago because I was afraid something like this would happen.

I urge you to pip install to 2.2.81,  

https://github.com/futzu/threefive/compare/v.2.2.45...master

If you have any issues, let me know and we'll resolve them quickly.

I appreciate the feedback. 

Adrian

**Reply by futzu** [reply]
https://github.com/futzu/threefive/commit/5e77cc24bede0905e057dafe2d5c7049ea0a39a5

**Reply by AudioDan** [reply]
Beautiful.  Thank you.  We are in the middle of a few tests during which we can't change anything, but looking forward to bumping to your next version soon.  Thanks again for the great product.

**Reply by futzu** [reply]
Boom goes the dynamite. ....

https://github.com/futzu/bitn/blob/79d233b1573f6cebc505e1182692acd399d117cc/bitn.py#L16

**Comment by futzu** [comment]
No more dependencies.

**Reply by AudioDan** [reply]
Nice!  We are off on another project temporarily, but will get back to this soon!

Thanks,

**Reply by futzu** [reply]
I've started using conditional imports for outside software.
```
ENCODE = True
try:
    import crcmod.predefined
except:
    ENCODE = False
```
